File: rag-backend/src/models/query.py
from pydantic import BaseModel

from models.activity import ActivityMetadata
from models.enums import InformationTier, RagPipeline, QuestionVariant
from models.profile import Profile

class QueryOptions(BaseModel):
    informationTier: InformationTier
    pipeline: RagPipeline
    useMaT: bool
    useESCOSkills: bool

class Query(BaseModel):
    profile: Profile
    query_id: int
    text_variants: dict[QuestionVariant, str]
    options: QueryOptions
    filter_values: ActivityMetadata | None

class QueryPacket(BaseModel):
    variants: dict[QuestionVariant, str]

class FusionRAGResponse(BaseModel):
    packets: list[dict[QuestionVariant, str]]

File: rag-web-ui/src/types/query.ts
import type { InformationTier, QuestionVariant, RagPipeline, Region, TimeFrame } from "./enums";
import type Profile from "./profile";

export interface Query {
    profile: Profile,
    query_id: number,
    text_variants: Record<QuestionVariant, string>
    options: QueryOptions
    filter_values: ActivityMetadata
}

export interface QueryOptions {
    informationTier: InformationTier;
    pipeline: RagPipeline;
    useMetadataFilter: boolean;
    useESCOSkills: boolean;
}

export interface RawQueryJson {
    id: number;
    text_variants: Record<QuestionVariant, string>;
}

export interface ActivityMetadata {
    region: Region,
    timeFrame: TimeFrame
}